### PR TITLE
Create documentation for InclusiveRange

### PR DESCRIPTION
**Pull request by dsainati1**

### CODE REVIEW

**Comment by dsainati1** [comment]
cc @darkdrag00nv2 please take a look at this and let me know how it looks

**Comment by dsainati1** [comment]
Ah I didn't realize this wasn't merged in yet. What are we waiting on to merge the feature branch?

**Comment by SupunS** [comment]
> What are we waiting on to merge the feature branch?

It was on hold till all the subtasks were completed, and then until the stable-cadence was merged to master.
Now that both are done, I started syncing it to master - there's quite a bit of conflicts there. Will open a PR soon.

File: versioned_docs/version-1.0/language/values-and-types.mdx [hunk 1]
```diff
@@ -1916,3 +1916,149 @@ Dictionary keys must be hashable and equatable.
...
+// start is 1, end is 100, step is 1
+let range: InclusiveRange<UInt> = InclusiveRange(1, 100)
+```
+
+Optionally a third, labeled, non-zero `step` argument can be provided to specify a step other than `1`. E.g., the following range contains 
+all odd numbers from 1 to 100:
+
+```cadence
+// start is 1, end is 100, step is 2
+let range: InclusiveRange<UInt> = InclusiveRange(1, 100, step: 2)
```
Comment (darkdrag00nv2) [inline versioned_docs/version-1.0/language/values-and-types.mdx:1942]: Same as above.

File: versioned_docs/version-1.0/language/values-and-types.mdx [hunk 2]
```diff
@@ -1916,3 +1916,149 @@ Dictionary keys must be hashable and equatable.
...
+
+    ```cadence
+    // Declare a range of `Int`s
+    let range: InclusiveRange<Int> = InclusiveRange(3, 9)
+
+    // Get the end of the range
+    let end = range.end
+
+    // `end` is `9`
+    ```
```
Comment (darkdrag00nv2) [inline versioned_docs/version-1.0/language/values-and-types.mdx:2009]: This seems to be repeated.

**Review by darkdrag00nv2 — COMMENTED** [review]
I believe we agreed that we should fix https://github.com/onflow/cadence/issues/2886 before shipping the feature.

See: https://github.com/onflow/cadence/issues/2886#issuecomment-1783839701

File: versioned_docs/version-1.0/language/values-and-types.mdx [hunk 3]
```diff
@@ -1916,3 +1916,149 @@ Dictionary keys must be hashable and equatable.
...
+A range value has a `start`, `end` and a `step`, which represents the interval at which the range's values are separated from `start` to `end`.  
+
+A range can be created using the `InclusiveRange` constructor, which can take two or three arguments. 
+
+In the case where the range is constructed with two arguments, the first argument is the `start` and the second is the `end`. 
+The `step` is inferred to be either `1` (when `end >= start`) or `-1` (when `end < start`). E.g.
+
+```cadence
+// start is 1, end is 100, step is 1
+let range: InclusiveRange<UInt> = InclusiveRange(1, 100)
```
[the remark this answers is not in the document]
Comment (dsainati1) [reply versioned_docs/version-1.0/language/values-and-types.mdx:1934]: With https://github.com/onflow/cadence/pull/3004 merged this should work now

**Comment by dsainati1** [comment]
@turbolent @SupunS now that the feature branch is merged this should be ready for review

**Pull request merged**